frame/evm/precompile/modexp/src/lib.rs

```diff
@@ -34,44 +34,52 @@ const MIN_GAS_COST: u64 = 200;
 
 // Calculate gas cost according to EIP 2565:
 // https://eips.ethereum.org/EIPS/eip-2565
-fn calculate_gas_cost(
-	base_length: u64,
-	exp_length: u64,
-	mod_length: u64,
-	exponent: &BigUint,
-) -> u64 {
+fn calculate_gas_cost(base_length: u64, mod_length: u64, exponent: &BigUint) -> u64 {
 	fn calculate_multiplication_complexity(base_length: u64, mod_length: u64) -> u64 {
 		let max_length = max(base_length, mod_length);
 		let mut words = max_length / 8;
 		if max_length % 8 > 0 {
 			words += 1;
 		}
 
-		// TODO: prevent/handle overflow
+		// Note: can't overflow because we take words to be some u64 value / 8, which is
+		// necessarily less than sqrt(u64::MAX).
+		// Additionally, both base_length and mod_length are bounded to 1024, so this has
+		// an upper bound of roughly (1024 / 8) squared
 		words * words
 	}
 
-	fn calculate_iteration_count(exp_length: u64, exponent: &BigUint) -> u64 {
+	fn calculate_iteration_count(exponent: &BigUint) -> u64 {
+		let exp_length_bytes = (exponent.bits() + 7) / 8;
```

Comment:
Why is this changed from `exp_length` instead? While this is technically closer to reflecting the actual "cost" of this precompile (`exp_length_bytes` are equal or smaller to `exp_length`), this doesn't look like what is specified in the EIP.